test/e2e/systemd_test.go

```diff
@@ -47,14 +47,21 @@ WantedBy=default.target
 	})
 
 	It("podman start container by systemd", func() {
-		SkipIfRootless("rootless can not write to /etc")
 		SkipIfContainerized("test does not have systemd as pid 1")
 
-		sysFile := os.WriteFile("/etc/systemd/system/redis.service", []byte(systemdUnitFile), 0644)
+		unitDir := "/run/systemd/system"
+		if isRootless() {
```

Comment:
This never triggers when I try to test on my laptop:
```console
$ ./ginkgo "podman start container by systemd"   <- basically invokes ~/goi/bin/ginkgo -focus
...
 Expected
      <*fs.PathError | 0xc00098f710>: {
          Op: "open",
          Path: "/run/systemd/system/redis.service",
          Err: <syscall.Errno>0xd,       <--- EACCESS
      }
  to be nil
```
Basically, `os.Geteuid()` is always returning `0`. I am not running this setuid or with sudo, and `~/go/bin/ginkgo` is not setuid. (If it was, I wouldn't be getting permission errors anyway).

I must be missing something very, very obvious. I feel very stupid for asking. But I'll ask anyway. Help please?

Reply:
The reason this is not working is because you are in the podman user namespace.
An import of pkg/rootless somewhere in the test will automatically join the user namespace.

Two ways to fix this: a) make sure ginkgo is started with the `remote` build tag. This turns the join code into a NOP.
b) kill the podman pause process (`catatonit -P`), this will not trigger the fast join path, in this case the code needs to explicitly create the namespace first.

Obviously a is the only proper solution which could look like this:
```diff
diff --git a/Makefile b/Makefile
index 672401d3c..f0dbc73b0 100644
--- a/Makefile
+++ b/Makefile
@@ -539,7 +539,7 @@ test: localunit localintegration remoteintegration localsystem remotesystem  ##
 .PHONY: ginkgo-run
 ginkgo-run:
        ACK_GINKGO_RC=true ginkgo version
-       ACK_GINKGO_RC=true ginkgo -v $(TESTFLAGS) -tags "$(TAGS)" $(GINKGOTIMEOUT) -cover -flakeAttempts 3 -progress -trace -noColor -nodes $(GINKGONODES) -debug $(GINKGOWHAT) $(HACK)
+       ACK_GINKGO_RC=true ginkgo -v $(TESTFLAGS) -tags "$(TAGS) remote" $(GINKGOTIMEOUT) -cover -flakeAttempts 3 -progress -trace -noColor -nodes $(GINKGONODES) -debug $(GINKGOWHAT) $(HACK)
 
 .PHONY: ginkgo
 ginkgo:
```

Note it does not matter if the tests run remote or local, this build tag is used by ginkgo and always required to prevent this bug.

Reply:
That was it! And, sigh, it exposed a bug in my rootless "fix". I'll pull that and save it for a rainy day.

Thank you!

Reply:
Running with `-tags remote` makes ginkgo think it's remote, so the `SkipIfRemote()` tests get skipped. How does this work in CI?

Reply:
Ahh, well in this case we need two different build tags then. This will require some more changes, I take a stab later today.

Reply:
https://github.com/containers/podman/pull/16322